### DISCUSSION TITLE
graph is cutted of - is it a bug?

### DISCUSSION THREAD

**Discussion by mrQanderson** [bloomberg/memray#488] [Q&A]
Hi all!

Could you please, kindly explain me.
Sometimes I noticed that my graph is placed only on some part of the page. And heap size memory is much more high than usual (in 2 times more). But if I rerun the CI job in Gitlab where test process is setup on shared runner - then could be fine. 
I attached the original HTML file.

[grapth_artifacts.zip](https://github.com/bloomberg/memray/files/13226344/grapth_artifacts.zip)
<img width="1800" alt="Screenshot 2023-11-01 at 14 12 38" src="https://github.com/bloomberg/memray/assets/66297758/95eec03f-0767-45aa-95b3-d3200d67638f">

**Comment by pablogsal** [comment]
This situation means that there are likely some native (non python) allocations happening under the main function. As by default these are not tracked the stack for them corresponds to the latest known python frame, in this case, main. This means that these allocation are likely happening in native threads. 

To show them you need to run memray with --native.

**Reply by mrQanderson** [reply]
Thank you so much, @pablogsal , for your attention and quick answer.

I attached below 2 html results in archive.
The same script was run at the same time, but in two independent shared runners in parallel. The same docker image in Gitlab CI shared runner. The same configuration of shared runner. The same packages except of one which new version I trying to test to find any memory leaks.
So the difference only in logic of one package under the hood.

memray 1.10
python 3.11

runtime instructions:
git clone --single-branch --branch ${RD_BRANCH} https://gitlab-ci-token:${CI_JOB_TOKEN}@gitlab.{company-name}.com/{project_name}/python-libraries/refinitiv-data.git
pip install --upgrade pip && pip install ./refinitiv-data/ memray && pip list
export PYTHONMALLOC="malloc"
memray run -o test_open_pricing_stream_iters.bin --trace-python-allocators scenarios/check_streaming_memory_leak/test_open_pricing_stream_iters.py
memray flamegraph -o test_open_pricing_stream_iters.html --leaks test_open_pricing_stream_iters.bin

The way how I try to determine leaks is to compare values right after graph line is fall down - first point and almost the last one or before last one. 
<img width="1091" alt="Screenshot 2023-11-02 at 06 04 25" src="https://github.com/bloomberg/memray/assets/66297758/25610838-0a4e-45e3-8736-2c73f1eced9a">

[Archive.zip](https://github.com/bloomberg/memray/files/13233803/Archive.zip)

Is anything I can do to avoid such affect to have clear results to determine leaks? Because it happening not always, but from time to time.
